integration/container/exec_test.go

```diff
@@ -106,6 +113,145 @@ func TestExec(t *testing.T) {
 	assert.Check(t, is.Contains(out, "FOO=BAR"), "exec command not running with expected environment variable FOO")
 }
 
+func TestExecResize(t *testing.T) {
+	ctx := setupTest(t)
+	apiClient := testEnv.APIClient()
+
+	cID := container.Run(ctx, t, apiClient, container.WithTty(true))
+	defer container.Remove(ctx, t, apiClient, cID, containertypes.RemoveOptions{Force: true})
+
+	cmd := []string{"top"}
+	if runtime.GOOS == "windows" {
+		cmd = []string{"sleep", "240"}
+	}
+	resp, err := apiClient.ContainerExecCreate(ctx, cID, containertypes.ExecOptions{
+		Tty:    true, // Windows requires a TTY for the resize to work, otherwise fails with "is not a tty: failed precondition", see https://github.com/moby/moby/pull/48665#issuecomment-2412530345
+		Detach: true,
+		Cmd:    cmd,
+	})
+	assert.NilError(t, err)
+	execID := resp.ID
+	assert.NilError(t, err)
+	err = apiClient.ContainerExecStart(ctx, execID, containertypes.ExecStartOptions{Detach: true})
+	assert.NilError(t, err)
+
+	t.Run("success", func(t *testing.T) {
+		err := apiClient.ContainerExecResize(ctx, execID, containertypes.ResizeOptions{
+			Height: 40,
+			Width:  40,
+		})
+		assert.NilError(t, err)
+		// TODO(thaJeztah): also check if the resize happened
+		//
+		// Note: container inspect shows the initial size that was
+		// set when creating the container. Actual resize happens in
+		// containerd, and currently does not update the container's
+		// config after running (but does send a "resize" event).
+	})
+
+	t.Run("invalid size", func(t *testing.T) {
+		const valueNotSet = "unset"
+
+		sizes := []struct {
+			doc, height, width, expErr string
+		}{
+			{
+				doc:    "unset height",
+				height: valueNotSet,
+				width:  "100",
+				expErr: `strconv.Atoi: parsing "": invalid syntax`,
```

Comment:
out of scope of this PR, but would be nice to have a better message here 😅

Reply:
Working on that exactly!